nodepool/scripts/convert_node_to_xenserver.sh

```diff
@@ -248,22 +248,24 @@ EOF
 }
 
 function print_postinst_file {
-    local rclocal
-    rclocal="$1"
+    local ci_rc_script
+    ci_rc_script="$1"
 
+    # ensure the ci_rc_script to be executed at the last of run-level 3.
     cat << EOF
 #!/bin/sh
 touch \$1/tmp/postinst.sh.executed
-cp \$1/etc/rc.d/rc.local \$1/etc/rc.d/rc.local.backup
-cat $rclocal >> \$1/etc/rc.d/rc.local
+cat $ci_rc_script >> \$1/etc/init.d/ci_rc_script
```

Comment:
Would it work to have a redirection to a nohup script?  
http://stackoverflow.com/questions/8124345/call-to-daemon-in-a-etc-init-d-script-is-blocking-not-running-in-background suggests that if you call:
       nohup ${exe} >/dev/null 2>&1 &
then it might work?